src/com/service/PersonService.java

```diff
@@ -0,0 +1,78 @@
+package com.service;
+
+import com.person.Person;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class PersonService implements IService{
+
+    private List<Person> allPersons;
+
+    public PersonService() {
+        this.allPersons = new ArrayList<>();
+    }
+
+    private boolean findPersonByDateOfBirth(Date dateOfBirth) {
+        for(int i=0; i<this.allPersons.size(); i++) {
+            Person existedPerson = this.allPersons.get(i);
+            if(existedPerson.getDateOfBirth() == dateOfBirth) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean isPersonCreated(Person person) {
+        if(findPersonByDateOfBirth(person.getDateOfBirth())) {
```

Comment:
You have the whole person with every name, status etc.. and you use just the date of birth here. Why ? What if there are 2 persons with the same date in the list what happens then? Here is a good example of why equals and hashcode should be overriden. Use it to compare two persons

Reply:
such a school boy, am I? Fixed that